apiserver/apiserver.go

```diff
@@ -876,10 +878,12 @@ func (srv *Server) endpoints() ([]apihttp.Endpoint, error) {
 		ctxt:          httpCtxt,
 		stateAuthFunc: httpCtxt.stateForMigrationImporting,
 	}, "tools")
-	resourcesMigrationUploadHandler := srv.monitoredHandler(&resourcesMigrationUploadHandler{
-		resourceServiceGetter:    &migratingResourceServiceGetter{ctxt: httpCtxt},
-		applicationServiceGetter: migratingApplicationServiceGetter,
-	}, "applications")
+	resourcesMigrationUploadHandler := srv.monitoredHandler(handlersresource.NewResourceMigrationUploadHandler(
+		logger,
```

Comment:
Logger should be last in the arguments, as it's the one you care the least about (same as above).